### PR TITLE
refactor: enable TCC using official GitHub action

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR re-enables TCC now that we've reduced container load _and_ a load balancing bug was fixed on their side.

I took the liberty of speeding up Keycloak and ES containers as well. As they both just need to spin up fast but not really do much, disabling GC logging and other system (note: not application, but system) logs is fine, and disabling JIT optimizations as well. This results in faster startup (ES by half, Keycloak a little less). At the same, I've switched Keycloak to use the official Testcontainers module as they suggested, which has a slightly different wait strategy. At the same time, I increased their startup time, as it's quite rare these will simply fail (and if they do crash, we would know immediately).

## Related issues

supersedes #13519

## Definition of Done

_Not all items need to be done depending on the issue and the pull request._

Code changes:
* [ ] The changes are backwards compatibility with previous versions
* [ ] If it fixes a bug then PRs are created to [backport](https://github.com/camunda/zeebe/compare/stable/0.24...main?expand=1&template=backport_template.md&title=[Backport%200.24]) the fix to the last two minor versions. You can trigger a backport by assigning labels (e.g. `backport stable/1.3`) to the PR, in case that fails you need to create backports manually.

Testing:
* [ ] There are unit/integration tests that verify all acceptance criterias of the issue
* [ ] New tests are written to ensure backwards compatibility with further versions
* [ ] The behavior is tested manually
* [ ] The change has been verified by a QA run
* [ ] The impact of the changes is verified by a benchmark

Documentation:
* [ ] The documentation is updated (e.g. BPMN reference, configuration, examples, get-started guides, etc.)
* [ ] If the PR changes how BPMN processes are validated (e.g. support new BPMN element) then the Camunda modeling team should be informed to adjust the BPMN linting.

Other teams:
If the change impacts another team an issue has been created for this team, explaining what they need to do to support this change.
- [ ] [Operate](https://github.com/camunda/operate/issues)
- [ ] [Tasklist](https://github.com/camunda/tasklist/issues)
- [ ] [Web Modeler](https://github.com/camunda/web-modeler/issues)
- [ ] [Desktop Modeler](https://github.com/camunda/camunda-modeler/issues)
- [ ] [Optimize](https://github.com/camunda/camunda-optimize/issues)

Please refer to our [review guidelines](https://github.com/camunda/zeebe/wiki/Pull-Requests-and-Code-Reviews#code-review-guidelines).

### CODE REVIEW

**Comment by npepinpe** [comment]
So the CI node itself can pull from the local registry. It seems to be something to do with the agent not picking up the private registry.

**Comment by npepinpe** [comment]
Not enabled yet for module integration tests, seems like all our ES tests are simply too heavy and ES ends up taking forever to spin up.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -80,13 +78,19 @@ jobs:
...
           distball: ${{ steps.build-zeebe.outputs.distball }}
-      - name: Prepare Testcontainers Cloud agent
-        if: env.TC_CLOUD_TOKEN != ''
-        run: |
-          curl -L -o agent https://app.testcontainers.cloud/download/testcontainers-cloud-agent_linux_x86-64
-          chmod +x agent
-          ./agent --private-registry-url=http://localhost:5000 '--private-registry-allowed-image-name-globs=*,*/*' --terminate > .testcontainers-agent.log 2>&1 &
-          ./agent wait
+      - name: Setup TCC
+        if: matrix.tcc-enabled == 'true'
```
Comment (npepinpe) [inline .github/workflows/ci.yml:82]: :thought_balloon: I was disappointed to see you can't just write `if: matrix.tcc-enabled` :( But maybe I'm missing something

File: exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java [hunk 2]
```diff
@@ -32,11 +34,20 @@ private TestSupport() {}
    *
    * <p>Additionally, security is explicitly disabled to avoid having tons of warning printed out.
    */
+  @SuppressWarnings("resource")
   static ElasticsearchContainer createDefaultContainer() {
     return new ElasticsearchContainer(ELASTIC_IMAGE)
-        .withEnv("ES_JAVA_OPTS", "-Xms256m -Xmx512m -XX:MaxDirectMemorySize=536870912")
+        // use JVM option files to avoid overwriting default options set by the ES container class
+        .withClasspathResourceMapping(
+            "elasticsearch-fast-startup.options",
```
Comment (npepinpe) [inline exporters/elasticsearch-exporter/src/test/java/io/camunda/zeebe/exporter/TestSupport.java:42]: :thought_balloon: ES has this nice way to allow users to customize JVM options without overwriting each other by using a folder with JVM option files.

File: qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java [hunk 3]
```diff
@@ -107,27 +109,14 @@ public class MultiTenancyOverIdentityIT {
...
+  private static final KeycloakContainer KEYCLOAK =
+      DefaultTestContainers.createDefaultKeycloak()
           .withEnv("KEYCLOAK_ADMIN_USER", KEYCLOAK_USER)
           .withEnv("KEYCLOAK_ADMIN_PASSWORD", KEYCLOAK_PASSWORD)
-          .withEnv("KEYCLOAK_DATABASE_HOST", DATABASE_HOST)
-          .withEnv("KEYCLOAK_DATABASE_PORT", String.valueOf(DATABASE_PORT))
-          .withEnv("KEYCLOAK_DATABASE_USER", DATABASE_USER)
-          .withEnv("KEYCLOAK_DATABASE_PASSWORD", DATABASE_PASSWORD)
-          .withEnv("KEYCLOAK_DATABASE_NAME", DATABASE_NAME)
+          .withEnv("KEYCLOAK_DATABASE_VENDOR", "dev-mem")
```
Comment (npepinpe) [inline qa/integration-tests/src/test/java/io/camunda/zeebe/it/multitenancy/MultiTenancyOverIdentityIT.java:116]: :thought_balloon: I honestly wasn't sure why they need Postgresql :shrug: That said, I would hope Identity could also work with H2 or some other in memory DB :(

File: qa/util/src/main/java/io/camunda/zeebe/qa/util/testcontainers/DefaultTestContainers.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.qa.util.testcontainers;
+
+import dasniko.testcontainers.keycloak.KeycloakContainer;
+import java.time.Duration;
+import org.testcontainers.utility.DockerImageName;
+
+public final class DefaultTestContainers {
```
Comment (npepinpe) [inline qa/util/src/main/java/io/camunda/zeebe/qa/util/testcontainers/DefaultTestContainers.java:14]: The idea is that for future containers which we reuse, we can share default config through this utility.

**Comment by npepinpe** [comment]
Re-assigning since meggle is busy with the workshop

**Comment by npepinpe** [comment]
Alright, I'm adding both, just hoping someone manages to review :sweat_smile:

**Review by lenaschoenburg — APPROVED** [review]
Nice :rocket: 

On first glance, IT is much faster :tada:

**Comment by npepinpe** [comment]
bors r-

Container based tests all failed lamentably

will look into it later

**Comment by npepinpe** [comment]
The weird part is all of them failed, so it sounds weird

**Comment by npepinpe** [comment]
Because the job was cancelled, I can't see the log, so I can't tell what happened. But every single container based test failed, so it's not a flakiness issue but an infrastructure problem.

**Comment by npepinpe** [comment]
Hm, so I previously tested it by running it multiple times without any issues. But today, all containers fail to start :thinking:

**Comment by npepinpe** [comment]
Alright, seems to be an issue on their side with the last release of their server binary :sweat_smile: I guess we're just unlucky. They're fixing it now :+1:

**Comment by npepinpe** [comment]
Interesting, all the backup tests are failing, which seem to indicate something weird with running the minio image :thinking:

**Comment by npepinpe** [comment]
So the issue was that TCC "released an optimization that should make your builds noticeable faster by using a tmpfs for volumes. But, due to misconfiguration, we had noexec flag set while mounting it, resulting it exec flag being masked out"

The new tmpfs mount also broke MinIO containers since we were using an old version from 2022. Bumping it should fix it.

**Comment by npepinpe** [comment]
bors merge

**Pull request merged**